pkg/schedule/coordinator.go

```diff
@@ -173,42 +188,106 @@ func (c *Coordinator) PatrolRegions() {
 		case <-ticker.C:
 			// Note: we reset the ticker here to support updating configuration dynamically.
 			ticker.Reset(c.cluster.GetCheckerConfig().GetPatrolRegionInterval())
+			newWorkersCount := c.cluster.GetCheckerConfig().GetPatrolRegionConcurrency()
+			if newWorkersCount != workersCount {
+				log.Info("coordinator starts patrol regions with new workers count",
+					zap.Int("old-workers-count", workersCount),
+					zap.Int("new-workers-count", newWorkersCount))
+				workersCount = newWorkersCount
+				close(quit)
+				wg.Wait()
+				quit = make(chan bool)
+				c.startPatrolRegionWorkers(workersCount, regionChan, quit, &wg)
+			}
+			if c.cluster.IsSchedulingHalted() {
+				for len(regionChan) > 0 {
+					<-regionChan
+				}
+				continue
+			}
+
+			// Check priority regions first.
+			c.waitDrainRegionChan(regionChan)
```

Comment:
Why do we need to drain the channel every time?

Reply:
Maybe I could add a map to ensure there is no the same region in this channel. Which do you think is the better option?